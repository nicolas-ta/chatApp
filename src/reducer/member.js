export const memberReducer = (state, action) => {
  switch (action.type) {
    case 'refresh': {
      const {members} = action.payload || {};
      return {...state, members, error: ''};
    }
    case 'add-member': {
      const {user} = action.payload || {};
      if (!state.members.map(m => m.userId).includes(user.userId)) {
        return {...state, members: [...state.members, user], error: ''};
      }
      break;
    }
    case 'remove-member': {
      const {user} = action.payload || {};
      if (state.members.map(m => m.userId).includes(user.userId)) {
        return {
          ...state,
          members: state.members.filter(m => m.userId !== user.userId),
          error: '',
        };
      }
      break;
    }
    case 'start-loading': {
      return {...state, loading: true, error: ''};
    }
    case 'end-loading': {
      const {error} = action.payload || {};
      return {...state, loading: false, error};
    }
    case 'error': {
      const {error} = action.payload || {};
      return {...state, error};
    }
  }
  return state;
};
