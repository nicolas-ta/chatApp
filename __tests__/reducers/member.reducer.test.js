import {memberReducer} from '@reducers/member';

describe('Member Reducer unit tests', () => {
  test('checking initial state', () => {
    const action = {type: 'nonexistent'};
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking refresh state', () => {
    const action = {type: 'refresh'};
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking add-member new state', () => {
    const action = {
      type: 'add-member',
      payload: {
        user: {
          userId: 'third_member',
        },
      },
    };

    const state = {
      members: [
        {
          userId: 'first_member',
        },
        {
          userId: 'second_member',
        },
      ],
    };

    expect(memberReducer(state, action)).toMatchSnapshot();
  });

  test('checking add-member existing state', () => {
    const action = {
      type: 'add-member',
      payload: {
        user: {
          userId: 'first_member',
        },
      },
    };

    const state = {
      members: [
        {
          userId: 'first_member',
        },
        {
          userId: 'second_member',
        },
      ],
    };

    expect(memberReducer(state, action)).toMatchSnapshot();
  });

  test('checking add-member missing argument state', () => {
    const action = {
      type: 'add-member',
      payload: {
        user: {
          userId: 'first_member',
        },
      },
    };

    const state = {
      members: [],
    };

    expect(memberReducer(state, action)).toMatchSnapshot();
  });

  test('checking add-member empty argument state', () => {
    const action = {
      type: 'add-member',
    };

    const state = {
      members: [],
    };

    expect(memberReducer(state, action)).toMatchSnapshot();
  });

  test('checking start-loading state', () => {
    const action = {type: 'start-loading'};
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking end-loading state', () => {
    const action = {type: 'end-loading'};
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });

  test('checking error state', () => {
    const action = {type: 'error'};
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });
});
