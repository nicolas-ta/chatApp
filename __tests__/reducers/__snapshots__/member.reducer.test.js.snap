// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Reducer unit tests checking add-member empty argument state 1`] = `
Object {
  "members": Array [],
}
`;

exports[`Member Reducer unit tests checking add-member existing state 1`] = `
Object {
  "members": Array [
    Object {
      "userId": "first_member",
    },
    Object {
      "userId": "second_member",
    },
  ],
}
`;

exports[`Member Reducer unit tests checking add-member missing argument state 1`] = `
Object {
  "error": "",
  "members": Array [
    Object {
      "userId": "first_member",
    },
  ],
}
`;

exports[`Member Reducer unit tests checking add-member new state 1`] = `
Object {
  "error": "",
  "members": Array [
    Object {
      "userId": "first_member",
    },
    Object {
      "userId": "second_member",
    },
    Object {
      "userId": "third_member",
    },
  ],
}
`;

exports[`Member Reducer unit tests checking end-loading state 1`] = `
Object {
  "error": undefined,
  "loading": false,
}
`;

exports[`Member Reducer unit tests checking error state 1`] = `
Object {
  "error": undefined,
}
`;

exports[`Member Reducer unit tests checking initial state 1`] = `undefined`;

exports[`Member Reducer unit tests checking refresh state 1`] = `
Object {
  "error": "",
  "members": undefined,
}
`;

exports[`Member Reducer unit tests checking start-loading state 1`] = `
Object {
  "error": "",
  "loading": true,
}
`;
